src/main/java/org/jvnet/hudson/test/RestartableJenkinsRule.java

```diff
@@ -34,7 +37,11 @@
 public class RestartableJenkinsRule implements MethodRule {
     public JenkinsRule j;
     private Description description;
-    private final List<Statement> steps = new ArrayList<Statement>();
+
+    /**
+     * List of {@link Statement}. For each one, the boolean value says if Jenkins is expected to start or not.
+     */
+    private final Map<Statement, Boolean> steps = new LinkedHashMap<>();
```

Comment:
🐜 I would declare the type here as *explicitly* LinkedHashmap since no other type could be legally used since we need ordering preserved.   

What we're doing here is a *bit* dodgy (using a LinkedHashmap to simulate a List of Tuple types, with no intent to really use mapping)... but it's also much less verbose than the "proper" alternative and doesn't require adding Apache Commons Lang [with a modern version to get tuple types](https://commons.apache.org/proper/commons-lang/apidocs/org/apache/commons/lang3/tuple/Pair.html).